prefect_gcp/cloud_storage.py

```diff
@@ -1140,3 +1214,71 @@ async def upload_from_folder(
         if num_uploaded == 0:
             self.logger.warning(f"No files were uploaded from {from_folder}.")
         return bucket_folder
+
+    @sync_compatible
+    async def upload_from_dataframe(
+        self,
+        df: DataFrame,
+        to_path: str,
+        output_format: Union[str, OutputFormat] = OutputFormat.CSV_GZIP,
+        **upload_kwargs: Dict[str, Any],
+    ) -> str:
+        """Upload a Pandas DataFrame to Google Cloud Storage in various formats.
+
+        This function uploads the data in a Pandas DataFrame to Google Cloud Storage
+        in a specified format, such as .csv, .csv.gz, .parquet,
+        .parquet.snappy, and .parquet.gz.
+
+        Args:
+            df (pd.DataFrame): The Pandas DataFrame to be uploaded.
+            to_path (str): The destination path for the uploaded DataFrame.
+            output_format (Union[str, OutputFormat]): The format to serialize
+                the DataFrame into. Defaults to `OutputFormat.CSV_GZIP`.
+                When passed as a `str`, the valid options are:
+                'csv', 'csv_gzip', 'parquet', 'parquet_snappy', 'parquet_gz'.
+            **upload_kwargs (Dict[str, Any]): Additional keyword arguments to
+                pass to the underlying `Blob.upload_from_dataframe` method.
+
+        Returns:
+            The path that the object was uploaded to.
+        """
+
+        def fix_extension_with(gcs_blob: str, ext: str):
+            """Fix the extension of a GCS blob.
+
+            Args:
+                gcs_blob (str): The path to the GCS blob to be modified.
+                ext (str): The extension to add to the filename.
+
+            Returns:
+                str: The modified path to the GCS blob with the new extension.
+            """
+            gcs_blob = Path(gcs_blob)
+            folder, filename = (gcs_blob.parent, Path(gcs_blob.stem).with_suffix(ext))
+            return str(folder.joinpath(filename))
+
+        if type(output_format) == str:
+            output_format = OutputFormat.lookup(output_format)
```

Comment:
```suggestion
            output_format = OutputFormat[output_format.upper()]
```

Reply:
Do you mean, this?

```python
output_format = OutputFormat.lookup(output_format.upper())
```

Reply:
I don't think we need the lookup method if I'm not mistaken.